components/navigation/navChild.js

```diff
@@ -65,24 +65,53 @@ const NavChild = ({ slug, page, color, className }) => {
   }
 
   let link;
+  let divider;
   let icon;
   let target;
-  let url = page.url;
+  let url;
+  if (typeof page.url !== "undefined") {
+    url = page.url;
+  } else {
+    url = "";
+  }
 
-  const isLocalPage = page.url.startsWith("/");
+  const isLocalPage = url.startsWith("/");
+  const isCrossLinkedPage = url.startsWith("https://docs.streamlit.io");
```

Comment:
Logic here can be faulty, since we could also have URLs that start with `www` or `http` if you want to live on the edge... Maybe we should check if the url includes `docs.streamlit.io` instead. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

Reply:
Variable are renamed and will show in next commit.

I was thinking of requiring the full HTTP path since it makes the later task of converting it to a relative link simpler. The consequence of getting this wrong with a later content update is just that it would look and behave like an external link and the symbol would make the mistake visually apparent.

I can change it if you think it's better to allow both versions of the URL.

Reply:
Got it! Yeah, I guess in order to fully automate this we should:
1. Create an array of potential absolute URLs;
2. Loop through it to see if any of those combinations (`http`, `https:` `www`, `https://www`, etc) fit the `page.url`;
3. If so, `.replace()` it;

Which might feel like a bit over-engineered for something like this. I'm fine with keeping it as it is, but maybe add a comment explaining why we require this specific format?

Reply:
I added a comment to the component.